### ISSUE TITLE
[Feature Request] Support for OpenAI Gym v0.22+

### ISSUE THREAD

**Issue by chrisyeh96** [DLR-RM/stable-baselines3#871]
Is there any estimated timeline for when OpenAI Gym v0.22+ will be supported?

gym v0.22 was understandably a large breaking change, but it would be great to know when SB3 might start supporting it.

**Comment by araffin** [comment]
duplicate of https://github.com/DLR-RM/stable-baselines3/issues/840

PS: if you want faster support, we need help to update our notebooks/tutorials

**Comment by araffin** [comment]
Gymnasium support (by default) is here: https://github.com/DLR-RM/stable-baselines3/pull/1327
With support for gym 0.21/0.26 and gymnasium env.

**Comment by araffin** [comment]
I just released an alpha version of the Gymnasium branch on PyPi:
```
# sb3 contrib installs sb3 too
pip install "sb3_contrib>=2.0.0a1" --upgrade
```
Documentation is available here: https://stable-baselines3.readthedocs.io/en/feat-gymnasium-support/

**Comment by ReHoss** [comment]
Hi @araffin ,

First thank everyone for the update to Gymnasium!

Could you or someone provide a shortlist of fixes to apply to our Gym 0.21 custom environments in order to keep reproducibility in the context of sb3 ?

Thank you for your consideration,
Best,

**Comment by araffin** [comment]
> Could you or someone provide a shortlist of fixes to apply to our Gym 0.21 custom environments in order to keep reproducibility in the context of sb3 ?

What do you mean exactly?
You want to keep using gym 0.21 or you want to upgrade to gymnasium and get the exact same results with SB3 as before when using the same setup (same machine, same os, same virtual env, same random seed)?

**Comment by ReHoss** [comment]
> > Could you or someone provide a shortlist of fixes to apply to our Gym 0.21 custom environments in order to keep reproducibility in the context of sb3 ?
> 
> What do you mean exactly? You want to keep using gym 0.21 or you want to upgrade to gymnasium and get the exact same results with SB3 as before when using the same setup (same machine, same os, same virtual env, same random seed)?

Yes I exactly want to **upgrade** to Gymnsasium while keeping same results when using same setup. I am just wondering if 0.21 custom envs adapt directly to 0.28 or do I need to fix some things?

**Comment by araffin** [comment]
> I am just wondering if 0.21 custom envs adapt directly to 0.28 or do I need to fix some things?

It might work out of the box (and you will need https://github.com/DLR-RM/stable-baselines3/pull/1486, we are missing reviewers for that PR) but I would say there are no warranty as Gymnasium changed a lot of things internally and might require a newer version of numpy.

As written in the documentation, we can only guaranty exact reproducibility for a given SB3/gym version. 
However, performance should be the same (when comparing quantitative evaluations).

**Comment by ReHoss** [comment]
> > I am just wondering if 0.21 custom envs adapt directly to 0.28 or do I need to fix some things?
> 
> It might work out of the box (and you will need #1486, we are missing reviewers for that PR) but I would say there are no warranty as Gymnasium changed a lot of things internally and might require a newer version of numpy.
> 
> As written in the documentation, we can only guaranty exact reproducibility for a given SB3/gym version. However, performance should be the same (when comparing quantitative evaluations).

Thanks, I'll provide some feedback at some point.
